test/IntegrationTests/StartupHookTests.cs

```diff
@@ -131,5 +132,14 @@ private void AssertExpectationsMet(List<WebServerSpanExpectation> expectations,
         string finalMessage = Environment.NewLine + string.Join(Environment.NewLine, failures.Select(f => " - " + f));
         Assert.True(!failures.Any(), finalMessage);
     }
+
+    private IImmutableList<IMockSpan> RunTestApplication(bool enableStartupHook = true)
+    {
+        int agentPort = TcpPortProvider.GetOpenPort();
+        using var agent = new MockZipkinCollector(Output, agentPort);
+        using var processResult = RunTestApplicationAndWaitForExit(agent.Port, enableStartupHook: enableStartupHook);
+
+        Assert.True(processResult.ExitCode >= 0, $"Process exited with code {processResult.ExitCode} and exception: {processResult.StandardError}");
+        return agent.WaitForSpans(2, TimeSpan.FromSeconds(5));
```

Comment:
It would be nice to have a method on the mock collector to get all spans already received and bail. This would avoid waiting for spans after the application already finished.

Reply:
Issue created https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/894